content/en/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md

```diff
@@ -0,0 +1,118 @@
+
+There are a few ways agents may run on kubernetes cluster. Agents may run on
+nodes directly or as DaemonSets.
+
+### Why do telemetry agents talk to Docker?
+
+Historically, kubernetes was built on top of Docker. Kubernetes is managing networking and scheduling, Docker was placing and operating containers on a node. So you can get scheduling-related metadata like a pod name from Kubernetes and containers state information from Docker. Over time more runtimes were created to manage containers. Also there are projects and kubernetes features that generalize container status information extraction across many runtimes.
+
+Some agents are tied specifically to the Docker tool. The agents may run commands like [`docker
+ ps`](https://docs.docker.com/engine/reference/commandline/ps/) or [`docker
+top](https://docs.docker.com/engine/reference/commandline/top/) to list
+containers and processes or [docker
+logs](https://docs.docker.com/engine/reference/commandline/logs/) to subscribe
+on docker logs. With the deprecating of Docker as a container runtime, these
+commands will not work any longer.
+
+
+### Identify DaemonSets that depend on Docker {#identify-docker-dependency}
+
+If a pod wants to make calls to the `dockerd` running on the node, the pod must either:
+
+- mount the filesystem containing the Docker daemon's privileged socket, as a
+  {{< glossary_tooltip text="volume" term_id="volume" >}}; or
+- mount the specific path of the Docker daemon's privileged socket directly, also as a volume.
+
+For example: on COS images, Docker exposes its Unix domain socket at
+`/var/run/docker.sock` This means that the pod spec will include a
+`hostPath` volume mount of `/var/run/docker.sock`.
+
+Here's a sample shell script to find Pods that have a mount directly mapping the
+Docker socket. This script outputs the namespace and name of the pod. You can
+remove the grep `/var/run/docker.sock` to review other mounts.
+
+```bash
+kubectl get pods --all-namespaces \
+-o=jsonpath='{range .items[*]}{"\n"}{.metadata.namespace}{":\t"}{.metadata.name}{":\t"}{range .spec.volumes[*]}{.hostPath.path}{", "}{end}{end}' \
+| sort \
+| grep '/var/run/docker.sock'
+```
+
+{{< note >}}
+There are alternative ways for a pod to access Docker on the host. For instance, the parent
+directory `/var/run` may be mounted instead of the full path (like in [this
+example](https://gist.github.com/itaysk/7bc3e56d69c4d72a549286d98fd557dd)).
+The script above only detects the most common uses.
+{{< /note >}}
+
+### Detecting Docker dependency from node deployed agents
+
+In case your cluster nodes are customized and install additional security and
+telemetry agents on the node, make sure to check with the vendor of the agent whether it has dependency on Docker.
+
+## Telemetry and security agent vendors
+
+This section is intended to collect information about various telemetry and
+security agents that may have dependency on containers runtime.
+
+### SignalFx (Splunk)
```

Comment:
I'm a little bit struggled at the section.
The overall guideline is that we are trying to avoid dual-hosted contents. However, it seems to me like we don't have a lot of options as of now ...

Reply:
I know. We can start with this and see whether vendors would love their own pages with instructions.

https://github.com/kubernetes/website/pull/25787#discussion_r548292025